bindings/go/mcp/mcp.go

```diff
@@ -92,12 +92,21 @@ func main() {
 
     var writer io.Writer
     if _, err := mio.ScanID(dst); err == nil {
+        if *pool != "" {
+            if _, err := mio.ScanID(*pool); err != nil {
+                log.Fatalf("invalid pool specified: %v", *pool)
+            }
+        }
         if err = mioW.Open(dst); err != nil {
             if err = mioW.Create(dst, objSize, *pool); err != nil {
                 log.Fatalf("failed to create object %v: %v", dst, err)
             }
         }
         defer mioW.Close()
+        if *pool != "" && !mioW.InPool(*pool) {
```

Comment:
Not clear why inversion (`!`) is needed for the `InPool()` return value. I would expect this condition to read like this:
```
if pool ID is not empty and object is present in that pool, then report an error "object already exist"
```

Reply:
@chumakd, no. You should read it: "if pool is not empty (i.e. was specified by user), but the object is *not* in that pool - report error".